sdk/tables/Azure.Data.Tables/tests/perf/Scenarios/AddComplexEntities.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Test.Perf;
+
+namespace Azure.Data.Tables.Performance
+{
+    public sealed class AddComplexEntities : TablesPerfTest
+    {
+        public AddComplexEntities(TablePerfOptions options) : base(options)
+        { }
+
+        public override async Task GlobalSetupAsync()
+        {
+            await base.GlobalSetupAsync().ConfigureAwait(false);
+        }
+
+        public override void Run(CancellationToken cancellationToken)
+        {
+            RunInternalAsync(false, cancellationToken).GetAwaiter().GetResult();
+        }
+
+        public override async Task RunAsync(CancellationToken cancellationToken)
+        {
+            await RunInternalAsync(true, cancellationToken).ConfigureAwait(false);
+        }
+
+        internal async Task RunInternalAsync(bool async, CancellationToken cancellationToken)
+        {
+            var entities = GenerateEntities<ComplexPerfEntity>();
+
+            foreach (var entity in entities)
+            {
+                if (async)
+                {
+                    await Client.AddEntityAsync(entity, cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
For the sake of efficiency, since you're awaiting in the `RunAsync` method,  have you considered not awaiting here and just returning the `Task`?    Same question for the other scenarios.

Reply:
I hadn't considered that - You mean collect the tasks and WaitAll on them? If never await the client calls, wouldn't RunAsync possibly complete before the client calls return?

Reply:
If I'm being honest, I overlooked the loop in this case....    It probably doesn't make sense here because the `WaitAll` would  introduce different behavior than the sync path. 

In the `ListXXX` scenarios where you're doing a single operation, I think skipping the await makese sens.